src/stream/stream_backend.rs

```diff
@@ -230,45 +230,51 @@ fn create_udp_stream(
             }
             _ => {
                 return Err(SimpleError::new(format!(
-                    "Unsupported GST source for UDP endpoint: {:#?}",
-                    gst_source
+                    "Unsupported GST source for UDP endpoint: {gst_source:#?}",
                 )));
             }
         },
         something => {
             return Err(SimpleError::new(format!(
-                "Unsupported VideoSourceType stream creation: {:#?}",
-                something
+                "Unsupported VideoSourceType stream creation: {something:#?}",
             )));
         }
     };
 
-    if VideoEncodeType::H264 == encode {
-        let udp_encode = concat!(
+    let video_payload = match &encode {
+        VideoEncodeType::H264 => concat!(
             " ! h264parse",
             " ! queue",
             " ! rtph264pay config-interval=10 pt=96",
-        );
+        ),
+        VideoEncodeType::YUYV => concat!(
+            " ! videoconvert",
+            " ! video/x-raw,format=UYVY",
```

Comment:
this is already the format on `video_format`, can you explain a bit more about this here ? can you also add a comment ? This looks more like an issue since you are converting from the same input as the output.

Reply:
Here we need to transcode from _YUYV_ (_YUY2_ on gstreamer), which is the raw video format from the source, to _UYVY_, which is one of the raw format accepted by the payloader _rtpvrawpay_, since _YUY2_ isn't.

```bash
~ gst-inspect-1.0 rtpvrawpay | grep "format:"
                 format: { (string)RGB, (string)RGBA, (string)BGR, (string)BGRA, (string)AYUV, (string)UYVY, (string)I420, (string)Y41B, (string)UYVP }
```

Should it be commented on in the code?

Reply:
It would be nice to document that since it's a bit hard to identify that reading the code.